handbook/ldap/ldap.md

```diff
@@ -0,0 +1,113 @@
+
+<figure>
+  <img src="../../img/ldap-setup.png" width="500" />
+  <figcaption>Configuration needed to enable logging into a SystemLink server using LDAP credentials.</figcaption>
+</figure>
+
+## LDAP URLs
+
+LDAP URLs follow a standard scheme. The URL determines the server to connect to, the server's port, a base search DN, and an attribute that determines what users will provide as a username when logging into SystemLink.
+
+!!! note "Example LDAP URL and login window"
+
+    **Structure:** `ldap://<server-dns>:<port>/<target-entry-dn>?<username-attribute>`
+
+    **Example:** `ldap://example.com:389/dc=example,dc=com?sAMAccountName`
+
+    In this example the server is `example.com`, the port is 389, the base search DN is `dc=example,dc=com` and the LDAP attribute used for user login is `sAMAccountName`. 
+
+    <figure>
+      <img src="../../img/ldap-login.png" width="500" />
+      <figcaption>Username `jdoe` is used because the `sAMAccountName` attribute is specified in the LDAP URL.</figcaption>
+    </figure>
+
+If a username attribute is not specified **NI Web Server Configuration** will automatically add `uid` as the attribute. Depending on how your LDAP directory is setup this attribute may or may not be available.
+
+!!! note "Common username attributes"
+    - `uid`
+    - `userPrincipalName`
+    - `sAMAccountName`
+
+## Bind User
+
+The bind user can be specified by providing either a `distinguishedName` or a `userPrincipalName`.
+
+!!! note "Bind user formats for the user Jane Doe"
+    **distinguishedName:** cn=jdoe,dc=example,dc=com
+
+    **userPrincipalName:** jane.doe@example.com
+
+## Mapping LDAP Attributes and Groups to SystemLink Workspaces and Roles
+
+To add a user or collection of users to a workspace and assign a role you must complete the role mapping workflow.
+
+1. Login into the SystemLink web application with a user mapped to the [**Server Administrator**](https://www.ni.com/documentation/en/systemlink/latest/setup/predefined-roles/) role.
+
+2. Go to **Security** > **Workspaces** and click the gear icon in the top right.
+
+3. Create a new workspace or edit an existing workspace.
+
+4. Go to the **Role Mappings** tab
+
+5. Click **+MAPPING** and select one of the available LDAP mapping types: *LDAP Group*, *LDAP User*, or *LDAP Attribute*.
+
+6. If you have have selected *LDAP Attribute* enter a valid key/value. For *LDAP User* or *LDAP Group* enter a valid value.
+
+### LDAP Group Mapping
+
+The LDAP Group mapping looks for the values in the LDAP `memberOf` attribute. This attribute's value is an array of group DNs. The value provided to SystemLink must exactly match one value available in the `memberOf` attribute array.
+
+### LDAP User Mapping
+
+Similar to the LDAP attribute that determines what users use for their username during login, the value for LDAP user mapping must match the user ID attribute specified in the LDAP URL.
```

Comment:
It took me a couple minutes to work out that the first clause here was referring back to a concept mentioned in the LDAP_URLs section. Also, "users use for their username" reads a bit awkward. I don't have a specific suggestion, but I think this needs to be clarified and likely broken up into more than one sentence.

Reply:
I also got hung up on this, thinking that the URL was something that the user had to enter in the mapping, when it's referring to the URL that was entered in the web server configuration.  Maybe this could say that the mapping value is the SystemLink login name and refer back to the other section.

A screenshot might help instead of the table (those URLs look like properly namespaced custom OpenID Connect Claims, which is also part of what confused me).

Reply:
Second this. Or maybe cut the first clause and just say 'LDAP user mappings must match the user ID attribute in the LDAP URL'?

I'm all for making connections in documentation, but in shorter sections like this it might just be distracting.

Reply:
I agree this felt wonky. I refactored each section some and I hope its more clear and succinct.